### DISCUSSION TITLE
How to create alarm based on resource don't have alarm methods

### DISCUSSION THREAD

**Discussion by DesAWSume** [cdklabs/cdk-monitoring-constructs#287] [Q&A] [answered]
Hi Awesome team! 

Very much obsessed with these awesome constructs you guys wrote, and it would be good to know how to add the alarm on the resource(EC2) that don't have alarm methods (for example **addCpuUsage**) based on your Features table on the front page https://github.com/cdklabs/cdk-monitoring-constructs

Objective:
Create a facade:
```
        monitoring = cdk_monitoring.MonitoringFacade(
            self,
            id="monitoring_facade",
            alarm_factory_defaults=cdk_monitoring.AlarmFactoryDefaults(
                action=cdk_monitoring.SnsAlarmActionStrategy(on_alarm_topic=alarm_topic),
                alarm_name_prefix=f'{props["service_name"]}',
                actions_enabled=True,
            ),
        )
```

define the resource - RDS
```
        monitoring.add_large_header("RDS-demo").monitor_rds_cluster(
            cluster=abc,
            add_cpu_usage_alarm={
                "Warning": cdk_monitoring.UsageThreshold(
                    datapoints_to_alarm=1,
                    evaluation_periods=1,
                    fill_alarm_range=True,
                    period=Duration.minutes(1),
                    max_usage_percent = 90
                    
                ),
                "Critical": cdk_monitoring.UsageThreshold(
                    datapoints_to_alarm=1,
                    evaluation_periods=1,
                    fill_alarm_range=True,
                    period=Duration.minutes(1),
                     max_usage_percent = 98
                ),                
            },
            add_disk_space_usage_alarm={
                "Critical": cdk_monitoring.UsageThreshold(
                    datapoints_to_alarm=1,
                    evaluation_periods=1,
                    period=Duration.minutes(1),
                    max_usage_percent = 98
                )
            }
        )
```

in contrast to EC2, it doesn't have the `add_disk_space_usage_alarm` & `add_cpu_usage_alarm` method build-in natively. 

```
        monitoring.add_large_header("EC2-monitoring").monitor_ec2_instances(
            instance_ids=[ajskas, aksjkas, aksjkasjaksj]
            add_to_detail_dashboard=True,
            add_to_alarm_dashboard=True,
            add_to_summary_dashboard=True,
        )
```

Can anyone tell me how to fully utlise this beat and share some light to me? 

Thanks all in advance.

**Comment by DesAWSume** [comment]
Anyone?

**Comment by ghost** [comment, marked as answer]
We'd encourage people to add support for any standard metrics and corresponding alarms as contributions. Check out [the contribution guide](https://github.com/cdklabs/cdk-monitoring-constructs/blob/main/CONTRIBUTING.md) if interested.

Looking at [the list of available standard EC2 metrics](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/viewing_metrics_with_cloudwatch.html), the `CPUUtilization` (which we already have in [the metric factory](https://github.com/cdklabs/cdk-monitoring-constructs/blob/aa33bc24e8dadc0b3dd09b2c4280040779164cb2/lib/monitoring/aws-ec2/EC2MetricFactory.ts#L155)) but I don't see something for disk space usage (although I might be missing it).

Otherwise, `monitorCustom` allows you to alarm on arbitrary metrics.
